pkg/gofr/config/godotenv.go

```diff
@@ -64,6 +66,14 @@ func (e *EnvLoader) read(folder string) {
 	} else {
 		e.logger.Infof("Loaded config from file: %v", overrideFile)
 	}
+
+	// Reload system environment variables to ensure they override any previously loaded values
+	for _, envVar := range os.Environ() {
+		keyValue := strings.SplitN(envVar, "=", splitParts)
+		if len(keyValue) == splitParts {
```

Comment:
There is a problem with the changes here, not the code but the logic, so maybe with the whole PR

Think about a password with `=` character in it

```dotenv
SECRET="foo="
```

Reply:
I have tested this scenario with updated implementation and it's working fine.

Reply:
Do you tji k you could add these tests to the PR then?